### PR TITLE
Drop Python 3.6 support

### LINKED ISSUE #34472: DEPS: drop 3.6
according to schedule we should 3.6 in june 2020

https://numpy.org/neps/nep-0029-deprecation_policy.html

prob ok to keep for the 1.1 release then should drop for 1.2. we will likely be adding. 3.9 support so this makes sense.

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #34472 
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
thanks, we would look to do this post 1.1 release, e.g. target 1.2 (so soonish)

File: setup.py [hunk 1]
```diff
@@ -766,7 +765,7 @@ def setup_package():
         long_description=LONG_DESCRIPTION,
         classifiers=CLASSIFIERS,
         platforms="any",
-        python_requires=">=3.6.1",
+        python_requires=">=3.7.1",
```
Comment (WillAyd) [inline setup.py:768]:
```suggestion
        python_requires=">=3.7.0",
```

I think the patch bump before was for some critical mypy fixes; we can probably start clean at 3.7

**Comment by VirosaLi** [comment]
The py37 minimum version CI times out after an hour. Not sure why

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=39023&view=logs&j=2d7fb38a-2053-50f3-a67c-09f6e91d3121&t=449937cc-3d50-56b5-5662-e489f41f1268

File: ci/deps/azure-37-minimum_versions.yaml [hunk 2]
```diff
@@ -15,16 +15,16 @@ dependencies:
...
+  - pytables=3.5.1
+  - python-dateutil=2.7.4
+  - pytz=2018.3
   - scipy=1.2
   - xlrd=1.1.0
-  - xlsxwriter=0.9.8
-  - xlwt=1.2.0
+  - xlsxwriter=1.0.2
+  - xlwt=1.3.0
   - html5lib=1.0.1
```
Comment (jorisvandenbossche) [inline ci/deps/azure-37-minimum_versions.yaml:30]: Are all those version increases in this file needed? (as this should test our minimal supported versions, generally, so if we need to update some, we might (depending on the reason) also need to update our listedd minimum versions)

Comment (VirosaLi) [reply ci/deps/azure-37-minimum_versions.yaml:30]: > Are all those version increases in this file needed? (as this should test our minimal supported versions, generally, so if we need to update some, we might (depending on the reason) also need to update our listedd minimum versions)

I double-checked again and lowered the versions for some of them in the new commit. Now they are the minimum versions without conflicts.

**Comment by VirosaLi** [comment]
I bumped the minimum python version to 3.7.1. That fixed the CI timeout. I've tested it on my local machine. While using 3.7.0, running this test class `pandas/tests/io/json/test_json_table_schema.py::TestTableOrient` will cause a Segmentation fault.

```bash
(pandas-test) fangchenli@fangchenli-home-office:~/Workspace/pandas-VirosaLi$ pytest pandas/tests/io/json/test_json_table_schema.py::TestTableOrient
============================================================================================================================================= test session starts =============================================================================================================================================
platform linux -- Python 3.7.0, pytest-5.0.1, py-1.9.0, pluggy-0.13.1
rootdir: /home/fangchenli/Workspace/pandas-VirosaLi, inifile: setup.cfg
plugins: forked-1.2.0, xdist-1.33.0, azurepipelines-0.8.0, hypothesis-5.19.1
collected 47 items                                                                                                                                                                                                                                                                                            

pandas/tests/io/json/test_json_table_schema.py ..........................................Fatal Python error: Segmentation fault

Current thread 0x00007f54ac64d740 (most recent call first):
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/sre_parse.py", line 166 in __getitem__
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/sre_parse.py", line 646 in _parse
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/sre_parse.py", line 426 in _parse_sub
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/sre_parse.py", line 816 in _parse
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/sre_parse.py", line 426 in _parse_sub
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/sre_parse.py", line 930 in parse
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/sre_compile.py", line 764 in compile
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/re.py", line 286 in _compile
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/re.py", line 234 in compile
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/dataclasses.py", line 196 in <module>
  File "<frozen importlib._bootstrap>", line 219 in _call_with_frames_removed
  File "<frozen importlib._bootstrap_external>", line 728 in exec_module
  File "<frozen importlib._bootstrap>", line 677 in _load_unlocked
  File "<frozen importlib._bootstrap>", line 967 in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 983 in _find_and_load
  File "/home/fangchenli/Workspace/pandas-VirosaLi/pandas/core/dtypes/inference.py", line 420 in is_dataclass
  File "/home/fangchenli/Workspace/pandas-VirosaLi/pandas/core/frame.py", line 500 in __init__
  File "/home/fangchenli/Workspace/pandas-VirosaLi/pandas/tests/io/json/test_json_table_schema.py", line 619 in test_timestamp_in_columns
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/python.py", line 165 in pytest_pyfunc_call
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/python.py", line 1451 in runtest
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/runner.py", line 117 in pytest_runtest_call
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/runner.py", line 192 in <lambda>
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/runner.py", line 220 in from_call
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/runner.py", line 192 in call_runtest_hook
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/runner.py", line 167 in call_and_report
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/runner.py", line 87 in runtestprotocol
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/runner.py", line 72 in pytest_runtest_protocol
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/main.py", line 278 in pytest_runtestloop
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/main.py", line 257 in _main
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/main.py", line 213 in wrap_session
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/main.py", line 250 in pytest_cmdline_main
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/callers.py", line 187 in _multicall
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 87 in <lambda>
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/manager.py", line 93 in _hookexec
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/pluggy/hooks.py", line 286 in __call__
  File "/home/fangchenli/miniconda3/envs/pandas-test/lib/python3.7/site-packages/_pytest/config/__init__.py", line 74 in main
  File "/home/fangchenli/miniconda3/envs/pandas-test/bin/pytest", line 11 in <module>
Segmentation fault (core dumped)
```
I haven't pinpointed why exactly this happened. But it does disappear for Python 3.7.1.

File: ci/azure/posix.yml [hunk 3]
```diff
@@ -9,40 +9,17 @@ jobs:
...
       ${{ if eq(parameters.name, 'Linux') }}:
-        py36_minimum_versions:
-          ENV_FILE: ci/deps/azure-36-minimum_versions.yaml
-          CONDA_PY: "36"
+        py37_minimum_versions:
+          ENV_FILE: ci/deps/azure-37-minimum_versions.yaml
+          CONDA_PY: "37"
           PATTERN: "not slow and not network and not clipboard"
 
-        py36_locale_slow_old_np:
```
Comment (alimcmaster1) [inline ci/azure/posix.yml:23]: We probably want to migrate these CI jobs to 37 opposed to removing

**Comment by VirosaLi** [comment]
Couldn't get the linux py38_np_dev ci to work. Need some help.

**Comment by simonjayhawkins** [comment]
needs a release note in doc\source\whatsnew\v1.2.0.rst detailing changes in minimum versions. see 'Increased minimum versions for dependencies' in doc\source\whatsnew\v1.0.0.rst

**Comment by simonjayhawkins** [comment]
> Couldn't get the linux py38_np_dev ci to work. Need some help.

these are maybe the known failures (should be 20 at this time, there are 29 here, merge upstream/master to make sure)

**Comment by VirosaLi** [comment]
I set the minimal python version to 3.7.1 for now. See #35309. We could lower it to 3.7.0 when we fixed that.

File: ci/azure/posix.yml [hunk 4]
```diff
@@ -9,57 +9,57 @@ jobs:
...
           PATTERN: "slow"
 
-        py36_locale:
-          ENV_FILE: ci/deps/azure-36-locale.yaml
-          CONDA_PY: "36"
-          PATTERN: "not slow and not network"
-          LANG: "it_IT.utf8"
-          LC_ALL: "it_IT.utf8"
-          EXTRA_APT: "language-pack-it xsel"
+#        py37_locale:
```
Comment (jreback) [inline ci/azure/posix.yml:38]: is there a reason these are commented?

File: ci/deps/azure-37-locale.yaml [hunk 5]
```diff
@@ -1,5 +1,6 @@
 name: pandas-dev
 channels:
+  - defaults
```
Comment (jreback) [inline ci/deps/azure-37-locale.yaml:3]: hmm why are you adding this?

**Review by jreback — CHANGES_REQUESTED** [review]
looks good.

**Comment by jreback** [comment]
merge master and see if can get the py38_np_dev build passing. ping on green. also have a couple of commented builds?

File: ci/deps/azure-37-locale.yaml (hunk 5 above)
Comment (VirosaLi) [reply ci/deps/azure-37-locale.yaml:3]: I think this one was originally azure-36-locale.yaml. I renamed it. Looks like GitHub didn't get this right. If you look at the side by side comparison for "ci/deps/azure-36-locale.yaml → ci/deps/azure-38-locale.yaml" below, the "-defaults" channel was there.

File: ci/azure/posix.yml (hunk 4 above)
Comment (VirosaLi) [reply ci/azure/posix.yml]: Travis has a py37_locale build. So I commented this out. I uncommented it in the last commit. But the CI test timed out. It looks similar to the issue mentioned in #35309.

File: ci/azure/posix.yml [hunk 6]
```diff
@@ -9,57 +9,57 @@ jobs:
...
           LANG: "it_IT.utf8"
           LC_ALL: "it_IT.utf8"
           EXTRA_APT: "language-pack-it xsel"
 
-        #py36_32bit:
-        #  ENV_FILE: ci/deps/azure-36-32bit.yaml
-        #  CONDA_PY: "36"
-        #  PATTERN: "not slow and not network and not clipboard"
-        #  BITS32: "yes"
+#        py37_32bit:
```
Comment (jreback) [inline ci/azure/posix.yml:46]: does this buidl not work?

File: ci/deps/azure-37-locale.yaml [hunk 7]
```diff
@@ -22,8 +23,11 @@ dependencies:
   - moto
   - nomkl
   - numexpr
-  - numpy
+  - numpy=1.16.*
   - openpyxl
+  # lowest supported version of pyarrow (putting it here instead of in
+  # azure-36-minimum_versions because it needs numpy >= 1.14)
```
Comment (jreback) [inline ci/deps/azure-37-locale.yaml:29]: hmm ok to bump to 0.15 which I thought we did already

File: ci/deps/azure-37-minimum_versions.yaml [hunk 8]
```diff
@@ -15,16 +15,16 @@ dependencies:
   # pandas dependencies
   - beautifulsoup4=4.6.0
   - bottleneck=1.2.1
-  - jinja2=2.8
+  - jinja2=2.10
   - numba=0.46.0
-  - numexpr=2.6.2
-  - numpy=1.15.4
+  - numexpr=2.6.8
+  - numpy=1.16.0
```
Comment (jreback) [inline ci/deps/azure-37-minimum_versions.yaml:21]: i wouldn't test the .0 version, go ahead with 1.16.5

File: ci/deps/azure-macos-37.yaml [hunk 9]
```diff
@@ -19,7 +19,7 @@ dependencies:
   - matplotlib=2.2.3
   - nomkl
   - numexpr
-  - numpy=1.15.4
+  - numpy=1.16.0
```
Comment (jreback) [inline ci/deps/azure-macos-37.yaml:22]: same

File: doc/source/getting_started/install.rst [hunk 10]
```diff
@@ -220,9 +220,9 @@ Dependencies
 Package                                                          Minimum supported version
 ================================================================ ==========================
 `setuptools <https://setuptools.readthedocs.io/en/latest/>`__    24.2.0
-`NumPy <https://www.numpy.org>`__                                1.15.4
+`NumPy <https://www.numpy.org>`__                                1.16.0
```
Comment (jreback) [inline doc/source/getting_started/install.rst:223]: use 1.16.5

File: doc/source/whatsnew/v1.2.0.rst [hunk 11]
```diff
@@ -21,6 +21,84 @@ Other enhancements
...
+Increased minimum versions for dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some minimum supported versions of dependencies were updated (:issue:`35214`).
+If installed, we now require:
+
++-----------------+-----------------+----------+---------+
+| Package         | Minimum Version | Required | Changed |
++=================+=================+==========+=========+
+| numpy           | 1.16.0          |    X     |    X    |
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.0.rst:42]: 1.16.5

**Review by jreback — CHANGES_REQUESTED** [review]

File: ci/deps/azure-37-locale.yaml [hunk 12]
```diff
@@ -22,8 +23,11 @@ dependencies:
...
+  - pyarrow=0.13
```
Comment (jorisvandenbossche) [inline ci/deps/azure-37-locale.yaml:30]: Can you check if you can put this now in minimum_versions? (see comment)

File: ci/azure/posix.yml (hunk 6 above)
Comment (VirosaLi) [reply ci/azure/posix.yml]: Not sure, I just uncommented it to see if it works.

Comment (VirosaLi) [reply ci/azure/posix.yml]: It failed. I'll keep this commented for now.

**Comment by VirosaLi** [comment]
The test `pandas\tests\io\parser\test_common.py:test_chunks_have_consistent_numerical_type` produces `ResourceWarning` occasionally on Windows-py37-np16 build. I skipped this test for windows.

On the NumPy dev build. The following tests:
```bash
FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_to_json
FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_date_format_raises
FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_datetime_tz
FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts0]
FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts1]
FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts2]
FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range0]
FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range1]
FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range2]
```

produce `DeprecationWarning: an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.`

I filtered this warning for now.

I'll open separate issues for those two warnings after this PR is merged.

**Comment by jreback** [comment]
> Those two tests
> 
> ```
> pandas\tests\io\parser\test_common.py:test_chunks_have_consistent_numerical_type
> pandas\tests\io\parser\test_common.py:test_warn_if_chunks_have_mismatched_type
> ```
> 
> produces `ResourceWarning` occasionally on Windows-py37-np16 and travis-37. I skipped them.
> 
> On the NumPy dev build. The following tests:
> 
> ```shell
> FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_to_json
> FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_date_format_raises
> FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_datetime_tz
> FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts0]
> FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts1]
> FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts2]
> FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range0]
> FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range1]
> FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range2]
> ```
> 
> produce `DeprecationWarning: an integer is required (got type float). Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.`
> 
> I filtered this warning for now.
> 
> I'll open separate issues for those two warnings after this PR is merged.

@VirosaLi thanks.

so looks all good. what is the status of the 32-bit build?

**Comment by VirosaLi** [comment]
> > Those two tests
> > ```
> > pandas\tests\io\parser\test_common.py:test_chunks_have_consistent_numerical_type
> > pandas\tests\io\parser\test_common.py:test_warn_if_chunks_have_mismatched_type
> > ```
> > 
> > 
> > produces `ResourceWarning` occasionally on Windows-py37-np16 and travis-37. I skipped them.
> > On the NumPy dev build. The following tests:
> > ```shell
> > FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_to_json
> > FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_date_format_raises
> > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_datetime_tz
> > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts0]
> > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts1]
> > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts2]
> > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range0]
> > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range1]
> > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range2]
> > ```
> > 
> > 
> > produce `DeprecationWarning: an integer is required (got type float). Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.`
> > I filtered this warning for now.
> > I'll open separate issues for those two warnings after this PR is merged.
> 
> @VirosaLi thanks.
> 
> so looks all good. what is the status of the 32-bit build?

It failed. So I left it commented. We could open another issue for the 32-bit build.

**Comment by jreback** [comment]
> > > Those two tests
> > > ```
> > > pandas\tests\io\parser\test_common.py:test_chunks_have_consistent_numerical_type
> > > pandas\tests\io\parser\test_common.py:test_warn_if_chunks_have_mismatched_type
> > > ```
> > > 
> > > 
> > > produces `ResourceWarning` occasionally on Windows-py37-np16 and travis-37. I skipped them.
> > > On the NumPy dev build. The following tests:
> > > ```shell
> > > FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_to_json
> > > FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_date_format_raises
> > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_datetime_tz
> > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts0]
> > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts1]
> > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts2]
> > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range0]
> > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range1]
> > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range2]
> > > ```
> > > 
> > > 
> > > produce `DeprecationWarning: an integer is required (got type float). Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.`
> > > I filtered this warning for now.
> > > I'll open separate issues for those two warnings after this PR is merged.
> > 
> > 
> > @VirosaLi thanks.
> > so looks all good. what is the status of the 32-bit build?
> 
> It failed. So I left it commented. We could open another issue for the 32-bit build.

do you recall how much work to get it running?

ok on opening an issue, can you followup on that build?

**Comment by jreback** [comment]
@TomAugspurger any objections to merging then doing the 32-bit build after?

**Comment by VirosaLi** [comment]
> > > > Those two tests
> > > > ```
> > > > pandas\tests\io\parser\test_common.py:test_chunks_have_consistent_numerical_type
> > > > pandas\tests\io\parser\test_common.py:test_warn_if_chunks_have_mismatched_type
> > > > ```
> > > > 
> > > > 
> > > > produces `ResourceWarning` occasionally on Windows-py37-np16 and travis-37. I skipped them.
> > > > On the NumPy dev build. The following tests:
> > > > ```shell
> > > > FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_to_json
> > > > FAILED pandas/tests/io/json/test_json_table_schema.py::TestTableOrient::test_date_format_raises
> > > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_datetime_tz
> > > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts0]
> > > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts1]
> > > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_is_utc[ts2]
> > > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range0]
> > > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range1]
> > > > FAILED pandas/tests/io/json/test_pandas.py::TestPandasContainer::test_tz_range_is_utc[tz_range2]
> > > > ```
> > > > 
> > > > 
> > > > produce `DeprecationWarning: an integer is required (got type float). Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.`
> > > > I filtered this warning for now.
> > > > I'll open separate issues for those two warnings after this PR is merged.
> > > 
> > > 
> > > @VirosaLi thanks.
> > > so looks all good. what is the status of the 32-bit build?
> > 
> > 
> > It failed. So I left it commented. We could open another issue for the 32-bit build.
> 
> do you recall how much work to get it running?
> 
> ok on opening an issue, can you followup on that build?

I'm not familiar with the 32-bit build. If I remember it correctly, it has been commented out for a while. I could work on it later this week.

**Comment by TomAugspurger** [comment]
I'm OK with dropping the 32-bit build. We need to restructure how it's done
anyway (since Anaconda isn't building 32-bit packages anymore).

**Review by jreback — APPROVED** [review]
ok then lets merge

**Comment by jreback** [comment]
thanks @VirosaLi very nice! i know this was a lot of effort

if you wouldn't mind opening the issues would be great

File: pandas/tests/io/parser/test_common.py [hunk 13]
```diff
@@ -1151,6 +1152,7 @@ def test_chunks_have_consistent_numerical_type(all_parsers):
     assert result.a.dtype == float
 
 
+@pytest.mark.skip("unreliable test #35214")
```
Comment (jbrockmendel) [inline pandas/tests/io/parser/test_common.py:1155]: 1) should this be xfail?
2) i think the solution will end up involving using td.check_file_leaks in a lot of places.  Is there a non-verbose way to run it once at the module level?

Comment (VirosaLi) [reply pandas/tests/io/parser/test_common.py:1155]: 1. Yes, it should be xfail. I'll submit a PR to fix this.
2. Haven't thought about this. Might need some discussion.

**Pull request merged**